instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/ui/routeline/RouteLineLayersTest.kt

```diff
@@ -98,84 +98,102 @@ class RouteLineLayersTest : BaseTest<BasicNavigationViewActivity>(
                         "background"
                     )
                 )
-
                 assertEquals(
                     "mapbox-layerGroup-1-restricted",
                     style.styleLayers[topLevelRouteLayerIndex - 1].id
                 )
                 assertEquals(
-                    "mapbox-layerGroup-1-traffic",
+                    "mapbox-masking-layer-traffic",
                     style.styleLayers[topLevelRouteLayerIndex - 2].id
                 )
```

Comment:
Suppose the second leg is or contains a restricted road and unauthorized usage resulted in a penalty of some kind. If the restricted road is masked it won't be visible until the driver is on it and at that point it's too late to avoid it.  I thought it better to always have the restricted road visible so drivers can make decisions about it.

Reply:
> Suppose the second leg is or contains a restricted road and unauthorized usage resulted in a penalty of some kind. If the restricted road is masked it won't be visible until the driver is on it and at that point it's too late to avoid it.

This actually creates, in my opinion, a very undesirable side effect. In situations where legs do not overlap, there's no problem in either of the approaches, restrictions are always visible. However, if legs do overlap and the restriction is only applicable in one way, in the current proposal we could be drawing a restriction on top of the current leg of the route even though it only applies in the other direction, which could be very confusing. Also, the restriction won't be masked until the driver is on it, it will only be masked until the driver reaches the first waypoint, so in all cases they should have an opportunity to deviate.

Reply:
@cafesilencio this still stands. I do think that restrictions on upcoming legs should appear below the current leg at all time.

Reply:
I made some adjustments resulting in the entire restricted section displaying when the line is first drawn so that on route overview the entire route is accurately represented. When navigation starts all the active leg layers are above inactive layers.  See the `Inactive Route Leg Styling With Restrictions` activity in the QA app. for a demonstration.